### DISCUSSION TITLE
sBTC Research - Generating Signing Bitmap

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacks-archive/sbtc-developer-release#318] [Ideas]
This is a research discussion post meant to be edited with research notes surrounding generating the bitmap that represents the signing round approval for the signer set. Note that this terminology is likely wrong but this discussion is here to track that research once the question becomes clearer.

**Comment by AshtonStephens** [comment]
Currently under investigation by @setzeus.

**Reply by setzeus** [reply]
When reading through this my interpretation is the following:

1. When successfully inserted (either deposit/withdraw) an authorization-map should be created with the TXID as key & with at least one field as signer-set that holds a list of successful signers
2. When a signer signs, their vote is registered in the authorization-map & their principal is appended to the list of signers

It’s unclear though whether there are also additional maps required? Is it possible we need a map for UTXOs?

**Comment by setzeus** [comment]
As a follow up from meeting it's an unknown unknown whether Stacker Authorization Transaction requires:

A. A bitmap of all voters
B. A bitmap of all voters **and** their vote direction

**Comment by jcnelson** [comment]
The reason the signing bitmap needs both those who vote to accept and those who vote to reject is because the set of Stackers must prove that at least 70% of them by weight were online when the vote was taken.  This would be the _sum_ of those who accept and those who reject, by STX weight.  This is important, because a liveness failure on the Stackers' part disables their PoX payouts and stops peg-ins until Stackers can reach quorum.  We can't measure liveness if only the "yes" votes are recorded.

**Reply by xoloki** [reply]
Answered in #317 

https://github.com/stacks-network/sbtc/discussions/317#discussioncomment-7394495
